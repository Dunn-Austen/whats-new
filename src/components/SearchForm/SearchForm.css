.search-container {
  align-items: center;
  display: flex;
  justify-content: space-around;
  width: 80%;
}

.search-input {
  font-size: 18px;
  height: 30px;
  text-indent: 5px;
}

.search-btn {
  height: 30px;
  width: 90px;
}

.search-btn:hover {
  background-color: yellow;
  color: black;
}
