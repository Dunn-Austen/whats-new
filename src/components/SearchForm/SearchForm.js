import React from 'react';
import './SearchForm.css';

const SearchForm = (props) => {

  return(
    <section className="search-form">
      <h4 className="search-text">Search Articles by Keyword:</h4>
      <div className="search-container">
        <input className="search-input" placeholder="Keyword"></input>
        <button className="search-btn" onClick={() => props.searchArticlesByKeyword(document.querySelector('.search-input').value)}>Search</button>
      </div>
    </section>
  )
}

export default SearchForm;
