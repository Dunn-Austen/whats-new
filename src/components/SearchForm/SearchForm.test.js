import React from 'react';
import SearchForm from './SearchForm';
import { shallow } from 'enzyme';

describe('SearchForm', () => {

  it('should match the snapshot with all data correctly passed', () => {
    const wrapper = shallow(  <SearchForm
      searchArticlesByKeyword={jest.fn()}
      />);

    expect(wrapper).toMatchSnapshot();
  })
})
