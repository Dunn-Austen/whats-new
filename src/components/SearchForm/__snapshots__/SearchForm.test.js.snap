// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm should match the snapshot with all data correctly passed 1`] = `
<section>
  <input
    className="search-input"
    placeholder="Keyword"
  />
  <button
    className="search-btn"
    onClick={[Function]}
  >
    Search
  </button>
</section>
`;
